build_tools/cmake/iree_macros.cmake

```diff
@@ -66,7 +66,10 @@ function(iree_package_ns PACKAGE_NS)
   # the corresponding rule in:
   #   build_tools/bazel_to_cmake/bazel_to_cmake_targets.py
   # Some sub-trees form their own roots for package purposes. Rewrite them.
-  if(_RELATIVE_PATH MATCHES "^compiler/src/(.*)")
+  if(IREE_ENABLE_MACRO_FOR_EXTERNAL_PROJECT)
+    # Use relative path directly for the external project.
+    set(_PACKAGE "${_RELATIVE_PATH}")
```

Comment:
How about we add a branch for files that are not located under `${IREE_ROOT_DIR}`? Then there would be no need for a separate option.

We could also add another argument to `iree_cc_library` (and the other functions) that would let you specify an explicit namespace.